apps/nextjs-website/src/helpers/user.helper.ts

```diff
@@ -11,17 +12,27 @@ export const useUser = () => {
   const [user, setUser] = useState<DevPortalUser | null>(null);
 
   const checkUser = useCallback(() => {
-    Auth.currentAuthenticatedUser()
-      .then((user) => {
+    const run = async () => {
+      const user = await Auth.currentAuthenticatedUser().catch(() => {
         setLoading(false);
-        setAligned(true);
-        setUser(user);
-      })
-      .catch(() => {
-        setLoading(false);
-        setAligned(true);
         setUser(null);
+        return null;
       });
+
+      if (!user) {
+        return;
+      }
+
+      const subscriptions = await getUserWebinarSubscriptions(
+        user.attributes.email
+      ).catch(() => []);
+
+      setLoading(false);
+      setAligned(true);
+      setUser({ ...user, webinarSubscriptions: subscriptions });
+    };
+
+    run();
```

Comment:
There is some particular reason to create `run` function that is only call here in `checkUser` instad of something like this:
```
  const checkUser = useCallback(async () => {
    const user = await Auth.currentAuthenticatedUser().catch(() => {
      setLoading(false);
      setUser(null);
      return null;
    });

    if (!user) {
      return;
    }

    const subscriptions = await getUserWebinarSubscriptions(
      user.attributes.email
    ).catch(() => []);

    setLoading(false);
    setAligned(true);
    setUser({ ...user, webinarSubscriptions: subscriptions });
  }, []);
```